### PR TITLE
Fixes drawing outside of plot by adding axis var

### PR DESCRIPTION
**Pull request by xksteven**
Fixes drawing outside of plot by adding axis variable to roipoly.  The if event.inaxes only checks if it is not None which will be false inside of buttons.

### CODE REVIEW

File: roipoly/roipoly.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def deprecation(message):
 
 class RoiPoly:
 
-    def __init__(self, fig=None, color='b',
+    def __init__(self, fig=None, ax=None, color='b',
```
Comment (jdoepfert) [inline roipoly/roipoly.py:25]: could you add the new argument to the docstring as well?

**Comment by jdoepfert** [comment]
Thx for ur contribution.
When I run `examples/multi_roi_example.py`, start drawing a new ROI, and then, while drawing, hover over one of the buttons, the current ROI-endpoint will move to the upper left of the image axes. Any idea why?
image
Any idea how to fix this?

File: roipoly/roipoly.py (hunk 1 above)
Comment (xksteven) [reply roipoly/roipoly.py:25]: Just added.  The change slipped my notice.

**Comment by xksteven** [comment]
When hovering over the button, the button's location is something like (0.8,0.2) which translates to a point near the very top left of the image.  Because the button has it's own Axes the mouse move callback will attempt to draw the line to the new (x,y) position as it enters the button area.

The line moving to the top left behavior should be fixed with this pull request though.  I just tested it to make sure it was the case on my machine.

**Comment by jdoepfert** [comment]
thanks for ur contribution! will be merged after tests.
fixes #19

**Pull request merged**